packages/core/http/core-http-router-server-internal/src/request.ts

```diff
@@ -189,8 +199,9 @@ export class CoreKibanaRequest<
       // a mistake. In v19, the `RouteSettings` interface does have an `app`
       // property.
       xsrfRequired:
-        ((request.route.settings as RouteOptions).app as KibanaRouteOptions)?.xsrfRequired ?? true, // some places in LP call KibanaRequest.from(request) manually. remove fallback to true before v8
-      tags: request.route.settings.tags || [],
+        ((request.route?.settings as RouteOptions)?.app as KibanaRouteOptions)?.xsrfRequired ??
+        true, // some places in LP call KibanaRequest.from(request) manually. remove fallback to true before v8
```

Comment:
I wonder if we should create a follow-up issue to remove this

Reply:
Yeah, this should likely be simplified ihmo. I was just kinda scare to change any of the behavior.